packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1010,12 +1023,18 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this.messageAtLastSummary = metadata?.message;
 
         this._connected = this.context.connected;
-        this.remoteMessageProcessor = new RemoteMessageProcessor(new OpSplitter(chunks), new OpDecompressor());
 
-        this.handleContext = new ContainerFluidHandleContext("", this);
+        this.mc = loggerToMonitoringContext(ChildLogger.create(this.logger, "ContainerRuntime"));
 
-        this.mc = loggerToMonitoringContext(
-            ChildLogger.create(this.logger, "ContainerRuntime"));
+        const opSplitter = new OpSplitter(
+            chunks,
+            this.context.submitBatchFn,
```

Comment:
I think leveraging this.context.submitBatchFn is risky as it can be undefined for old loaders. I do not see code takes care of it. I'd expect tests against LTS branch to fail.
General direction here should be simplicity of the code, i.e. if we are dealing with old loader, just do not split (unless it's trivial to use old APIs as alternative)

Reply:
Good point.